### ISSUE TITLE
Directly inheriting from ActiveRecord::Migration is not supported. Please specify the Rails release the migration was written for:    class AddSessionsTable < ActiveRecord::Migration[4.2]

### ISSUE THREAD

**Issue by jasonfb** [rails/activerecord-session_store#138]
migration error on Rails 5.1, 5.2

```
Directly inheriting from ActiveRecord::Migration is not supported. Please specify the Rails release the migration was written for:

  class AddSessionsTable < ActiveRecord::Migration[4.2]
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:528:in `inherited'
/Users/jason/Work/Mobile-Notary/API-app-api.mobilemontarycenter.com/mobilenotarycenter_api/db/migrate/20181212121140_add_sessions_table.rb:1:in `<top (required)>'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activesupport-5.2.1/lib/active_support/dependencies.rb:287:in `block in require'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activesupport-5.2.1/lib/active_support/dependencies.rb:253:in `load_dependency'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activesupport-5.2.1/lib/active_support/dependencies.rb:287:in `require'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:986:in `load_migration'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:982:in `migration'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:977:in `disable_ddl_transaction'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1350:in `use_transaction?'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1342:in `ddl_transaction'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1291:in `execute_migration_in_transaction'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1263:in `block in migrate_without_lock'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1262:in `each'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1262:in `migrate_without_lock'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1210:in `block in migrate'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1363:in `with_advisory_lock'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1210:in `migrate'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1036:in `up'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1011:in `migrate'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/tasks/database_tasks.rb:172:in `migrate'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/railties/databases.rake:60:in `block (2 levels) in <top (required)>'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/rake-12.3.1/exe/rake:27:in `<top (required)>'
/Users/jason/.rvm/gems/ruby-2.2.5/bin/ruby_executable_hooks:24:in `eval'
/Users/jason/.rvm/gems/ruby-2.2.5/bin/ruby_executable_hooks:24:in `<main>'

Caused by:
StandardError: Directly inheriting from ActiveRecord::Migration is not supported. Please specify the Rails release the migration was written for:

  class AddSessionsTable < ActiveRecord::Migration[4.2]
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:528:in `inherited'
/Users/jason/Work/Mobile-Notary/API-app-api.mobilemontarycenter.com/mobilenotarycenter_api/db/migrate/20181212121140_add_sessions_table.rb:1:in `<top (required)>'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activesupport-5.2.1/lib/active_support/dependencies.rb:287:in `block in require'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activesupport-5.2.1/lib/active_support/dependencies.rb:253:in `load_dependency'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activesupport-5.2.1/lib/active_support/dependencies.rb:287:in `require'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:986:in `load_migration'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:982:in `migration'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:977:in `disable_ddl_transaction'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1350:in `use_transaction?'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1342:in `ddl_transaction'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1291:in `execute_migration_in_transaction'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1263:in `block in migrate_without_lock'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1262:in `each'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1262:in `migrate_without_lock'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1210:in `block in migrate'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1363:in `with_advisory_lock'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1210:in `migrate'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1036:in `up'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/migration.rb:1011:in `migrate'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/tasks/database_tasks.rb:172:in `migrate'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/activerecord-5.2.1/lib/active_record/railties/databases.rake:60:in `block (2 levels) in <top (required)>'
/Users/jason/.rvm/gems/ruby-2.2.5/gems/rake-12.3.1/exe/rake:27:in `<top (required)>'
/Users/jason/.rvm/gems/ruby-2.2.5/bin/ruby_executable_hooks:24:in `eval'
/Users/jason/.rvm/gems/ruby-2.2.5/bin/ruby_executable_hooks:24:in `<main>'
Tasks: TOP => db:migrate
```

**Comment by jasonfb** [comment]
same as closed https://github.com/rails/activerecord-session_store/issues/124

**Comment by jasonfb** [comment]
seems like this is core functionality to Rails and should be supported here. I recently did something similar for my own gem. Will work on PR

**Comment by sikachu** [comment]
This should be fixed in 1.1.2. Please let me know if you still running into any issue.
